src/partition/src/splitter.rs

```diff
@@ -327,6 +367,54 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_writer_spliter_when_insert_columns_len_is_not_equal_to_schema_columns_len() {
```

Comment:
We can safely remove this test because we already removed the code potential to cause panic.